hack/test-tekton-kueue-config.py

```diff
@@ -0,0 +1,486 @@
+        """Run a single mutation test and return results."""
+        pipelinerun = test_data["pipelinerun"]
+        
+        with tempfile.TemporaryDirectory() as temp_dir:
+            # Write the config file
+            config_path = Path(temp_dir) / "config.yaml"
+            pipelinerun_path = Path(temp_dir) / "pipelinerun.yaml"
+            
+            # Copy the config file
+            import shutil
+            shutil.copy2(CONFIG_FILE, config_path)
+            
+            # Write the PipelineRun
+            with open(pipelinerun_path, 'w') as f:
+                yaml.dump(pipelinerun, f, default_flow_style=False)
+            
+            # Set proper permissions
+            os.chmod(config_path, 0o644)
+            os.chmod(pipelinerun_path, 0o644)
+            os.chmod(temp_dir, 0o755)
+            
+            # Run the mutation
+            cmd = [
+                "podman", "run", "--rm",
+                "-v", f"{temp_dir}:/workspace:z",
+                self.tekton_kueue_image,
+                "mutate",
+                "--pipelinerun-file", "/workspace/pipelinerun.yaml",
+                "--config-dir", "/workspace"
+            ]
+            
+            result = subprocess.run(cmd, capture_output=True, text=True)
+            
+            if result.returncode != 0:
+                self.fail(f"Mutation failed: {result.stderr}")
+            
+            # Parse the mutated PipelineRun
+            try:
+                mutated = yaml.safe_load(result.stdout)
+            except yaml.YAMLError as e:
+                self.fail(f"Failed to parse mutated YAML: {e}")
+            
+            return mutated
+    
+    def validate_mutation_result(self, test_key: str, test_data: Dict):
+        """Helper method to validate mutation results."""
+        with self.subTest(test=test_key):
+            mutated = self.run_mutation_test(test_data)
+            expected = test_data["expected"]
+            
+            # Check annotations
+            annotations = mutated.get("metadata", {}).get("annotations", {})
+            for expected_annotation in expected["annotations"]:
+                self.assertIn(expected_annotation, annotations, 
+                             f"Expected annotation {expected_annotation} not found")
+                self.assertEqual(annotations[expected_annotation], "1",
+                               f"Expected annotation {expected_annotation} to have value '1'")
+            
+            # Check labels
+            labels = mutated.get("metadata", {}).get("labels", {})
+            for expected_label, expected_value in expected["labels"].items():
+                self.assertIn(expected_label, labels,
+                             f"Expected label {expected_label} not found")
+                self.assertEqual(labels[expected_label], expected_value,
+                               f"Expected label {expected_label} to have value '{expected_value}', got '{labels.get(expected_label)}'")
```

Comment:
it seems to me we can simplify this:

```suggestion
            self.assertDictEqual(mutated, expected)
```